java/src/hdf/hdf5lib/HDFArray.java

```diff
@@ -522,36 +518,34 @@ else if (ArrayDescriptor.className.equals("java.lang.Long")) {
             flattenedArray = (Object) _barray;
             break;
         case 'L':
-            switch (ArrayDescriptor.className) {
```

Comment:
Is this file maybe not up-to-date?

Reply:
I can understand the changes for this switch, again this should be extracted to a separate issue.